### PR TITLE
[No QA] Code cleanup / Add story for OptionRow

### PR DESCRIPTION
**Pull request by marcaaron**
### Details

- `OptionRow` is a fairly commonly used component, but for some reason was in `/sidebar/` directory instead of `components`
- There were also propTypes in that directory getting used in various places and another propTypes file ([optionPropTypes.js](https://github.com/Expensify/App/compare/marcaaron-optionRowStory?expand=1#diff-2ec1b0766b5971799916cc70545730bea1c801f32c8968fe075bbb1c5ce13226)) that shared many overlapping props with a file of the same name in `/components`.
- Consolidated everything, moved it all into `/components` and created a story for the `OptionRow` so it is easier to see when to use it vs. something like `MenuItem`

### Fixed Issues

No issue in particular just adding a story for OptionRow / cleaning stuff up

### Tests
`npm run storybook` and verify the `OptionRow` appears

- [x] Verify that no errors appear in the JS console

### QA Steps
❌

### CODE REVIEW

**Review by roryabraham — DISMISSED** [review]
This is all good cleanup, so LGTM 👍 . Whether or not you want to take the time to address the other comments is up to you.

File: src/components/optionPropTypes.js [hunk 1]
```diff
@@ -33,4 +34,16 @@ export default PropTypes.shape({
 
     // Whether the report corresponds to a chat room
     isChatRoom: PropTypes.bool,
+
```
[the remark this answers is not in the document]
Comment (marcaaron) [reply src/components/optionPropTypes.js:37]: Looks like we need to add it! On it!

File: src/components/OptionRow.js [hunk 2]
```diff
@@ -77,7 +77,7 @@ const defaultProps = {
     forceTextUnreadStyle: false,
     showTitleTooltip: false,
     mode: 'default',
-    onSelectRow: null,
+    onSelectRow: () => {},
     isDisabled: false,
```
[the remark this answers is not in the document]
Comment (marcaaron) [reply src/components/OptionRow.js:81]: Nice. I think that makes sense to me. `disableRowInteractivity` doesn't really fit our other naming conventions and `isDisabled` is a much better name.

**Review by amyevans — APPROVED** [review]
lgtm, verified the `OptionRow` appears and no JS errors in the console as well

**Pull request merged**